src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php

```diff
@@ -64,24 +64,133 @@ public static function castThrowingCasterException(ThrowingCasterException $e, a
+
+        for ($j -= $i++; isset($frames[$i]); ++$i, --$j) {
+            $call = isset($frames[$i]['function']) ? (isset($frames[$i]['class']) ? $frames[$i]['class'].$frames[$i]['type'] : '').$frames[$i]['function'].'()' : '???';
+
+            $a[Caster::PREFIX_VIRTUAL.$j.'. '.$call.$lastCall] = new FrameStub(
+                array(
+                    'object' => isset($frames[$i]['object']) ? $frames[$i]['object'] : null,
+                    'class' => isset($frames[$i]['class']) ? $frames[$i]['class'] : null,
+                    'type' => isset($frames[$i]['type']) ? $frames[$i]['type'] : null,
+                    'function' => isset($frames[$i]['function']) ? $frames[$i]['function'] : null,
+                ) + $frames[$i - 1],
+                $trace->srcContext,
+                $trace->keepArgs,
+                true
             );
-            array_splice($b[$xPrefix.'trace'], -1 - count($a[$xPrefix.'trace']));
-            static::filterTrace($b[$xPrefix.'trace'], false);
-            $a[Caster::PREFIX_VIRTUAL.'trace'] = $b[$xPrefix.'trace'];
+
+            $lastCall = ' ==> '.$call;
+        }
+        $a[Caster::PREFIX_VIRTUAL.$j.'. {main}'.$lastCall] = new FrameStub(
+            array(
+                'object' => null,
+                'class' => null,
+                'type' => null,
+                'function' => '{main}',
+            ) + $frames[$i - 1],
+            $trace->srcContext,
+            $trace->keepArgs,
+            true
+        );
+        if (null !== $trace->length) {
+            $a = array_slice($a, 0, $trace->length, true);
+        }
+
+        return $a;
+    }
+
+    public static function castFrameStub(FrameStub $frame, array $a, Stub $stub, $isNested)
+    {
+        if (!$isNested) {
+            return $a;
         }
+        extract($frame->value);
+        $prefix = Caster::PREFIX_VIRTUAL;
 
-        unset($a[$xPrefix.'trace'], $a[$xPrefix.'previous'], $a[$prefix.'code'], $a[$prefix.'file'], $a[$prefix.'line']);
+        if (isset($file, $line)) {
+            if (preg_match('/\((\d+)\)(?:\([\da-f]{32}\))? : (?:eval\(\)\'d code|runtime-created function)$/', $file, $match)) {
+                $file = substr($file, 0, -strlen($match[0]));
+                $line = (int) $match[1];
+            }
+            if (file_exists($file) && 0 <= $frame->srcContext) {
+                $src[$file.':'.$line] = self::extractSource(explode("\n", file_get_contents($file)), $line, $frame->srcContext);
+
+                if (!empty($class) && is_subclass_of($class, 'Twig_Template') && method_exists($class, 'getDebugInfo')) {
+                    $twig = isset($object) ? $object : new $class(new \Twig_Environment(new \Twig_Loader_Filesystem()));
+
+                    try {
+                        $twigName = $twig->getTemplateName();
+                        $twigSrc = explode("\n", method_exists($twig, 'getSource') ? $twig->getSource() : $twig->getEnvironment()->getLoader()->getSource($twigName));
```

Comment:
this is broken: the filesystem loader is not configured, and so will not be able to find the source.

And in recent Twig versions, the cache key also depends on the enabled extensions, making things even worse.

I suggets going the easy way: not getting the Twig source at all for older Twig versions. People wanting to get the nice output for Twig templates should just update Twig to the latest version.